cmake/AutoOptionHelpers.cmake

```
@@ -37,18 +37,26 @@ function(_REGISTER_AUTO_OPTION _NAME _FEATURE_VAR _DESCRIPTION _DEFAULT)
   )
 endfunction()
 
-macro(_CHECK_PACKAGE_DEPENDS _OPTION_VAR _PACKAGE_DEPENDS _FEATURE_VAR)
+macro(_CHECK_PACKAGE_DEPENDS _OPTION_VAR _PACKAGE_DEPENDS _FEATURE_VAR _COMPONENT)
   if(${${_OPTION_VAR}} STREQUAL AUTO)
     set(STRICTNESS)
   else()
     set(STRICTNESS REQUIRED)
   endif()
 
   foreach(PACKAGE_NAME ${_PACKAGE_DEPENDS})
-    find_package(${PACKAGE_NAME} ${STRICTNESS})
-    if(NOT ${PACKAGE_NAME}_FOUND)
-      set(${_FEATURE_VAR} FALSE)
-      break()
+    if(NOT ${_COMPONENT} STREQUAL OFF)
```

What about:
```suggestion
    if(${_COMPONENT})
```

`"OFF"`, `"0"`, `"FALSE"` and `""` should all have falsey values, I think.

---

I'm pretty sure I tried this and couldn't get it to work.